### PR TITLE
Rf drag and drop

### PR DESCRIPTION
**Pull request by bramalingam**
This PR adds tests to evaluate drag and drop functionality in the JS tree.

Testing Instructions:
Check if all Tests are passing.
@will-moore @aleksandra-tarkowska

### CODE REVIEW

File: components/tests/ui/resources/web/tree.txt [hunk 1]
```diff
@@ -187,6 +188,13 @@ Select First Dataset With Children
     Wait Until Right Panel Loads    Dataset                  ${datasetId}
     [Return]                        ${datasetId}
 
+Select First Datast With Name
```
Comment (aleksandra-tarkowska) [inline components/tests/ui/resources/web/tree.txt]: Dataset, you are not using this keyword below

File: components/tests/ui/testcases/web/post_test.txt [hunk 2]
```diff
@@ -88,11 +136,49 @@ Test Cut Paste Image
...
+    Select Experimenter
+    ${s1id}=                                Create Screen      test Cut-Paste TO here
+    Click Element                           refreshButton
+
+    ${s2id}=                                Select First Screen With Children
+    ${screenNode}                           Wait For Screen Node                 ${s2id}
+    ${plateId}=                             Select First Plate
+    Click Element                           id=cutButton    
+    
+    Wait Until Keyword Succeeds           10    ${INTERVAL}   Page Should Not Contain Element    xpath=//li[@id='${screenNode}']//li[@data-id='${plateId}']
```
Comment (aleksandra-tarkowska) [inline components/tests/ui/testcases/web/post_test.txt]: There is no need to hardcore 10. Use ${TIMEOUT}

File: components/tests/ui/testcases/web/post_test.txt [hunk 3]
```diff
@@ -114,3 +200,203 @@ Test Chgrp
...
+
+    Select Experimenter
+    ${did}=                                     Create Dataset
+    Click Element                               refreshButton
+
+    Select Experimenter
+    ${pId}    ${datasetId}    ${imageId}        Select And Expand Project Dataset Image
+    Wait Until Element is Visible               xpath=//li[@data-id='${datasetId}']//li[@data-id='${imageId}']/a
+
+    Sleep                                       0.5
```
Comment (aleksandra-tarkowska) [inline components/tests/ui/testcases/web/post_test.txt]: Is there necessary to use sleep? There should be keyword guarantee tree element appeared.

File: components/tests/ui/resources/web/tree.txt (hunk 1 above)
Comment (bramalingam) [inline components/tests/ui/resources/web/tree.txt]: Have fixed the typo here. And this keyword is being used in the Drag and Drop Negative tests

File: components/tests/ui/testcases/web/post_test.txt (hunk 3 above)
Comment (bramalingam) [inline components/tests/ui/testcases/web/post_test.txt]: The sleep was added as a precautionary measure, I've removed them now.

File: components/tests/ui/testcases/web/post_test.txt (hunk 2 above)
Comment (bramalingam) [inline components/tests/ui/testcases/web/post_test.txt]: True. Was going to update that throughout the test, once the build turns green.

**Comment by sbesson** [comment]
Temporarily excluding for this afternoon's rebuild.

**Comment by bramalingam** [comment]
Pulled snoopy/develop/merge/daily, and ran the tests locally on a fresh database setup and all tests pass.!

Still need to figure out the issue behind the database state, affecting tests..!
```
ls29010:openmicroscopy bramalingam$ ./build.py -f components/tests/ui/build.xml web-chrome -DTEST=post_test.txt
Buildfile: /Users/bramalingam/OME/openmicroscopy/components/tests/ui/build.xml
Entering /Users/bramalingam/OME/openmicroscopy/components/tests/ui...
web-chrome:
==============================================================================
Post Test :: Tests ajax 'POST' updates.                                       
==============================================================================
Test Cut Paste Image :: Cut an existing Image to /remove/ it to Or... | PASS |
------------------------------------------------------------------------------
Test Cut Paste Orphaned Image :: Cut an orphaned Image and paste i... | PASS |
------------------------------------------------------------------------------
Test Cut Paste Dataset :: Create 2 Projects and a Dataset. Cut and... | PASS |
------------------------------------------------------------------------------
Test Cut Paste Plate :: Cut a Plate and paste it onto another screen  | PASS |
------------------------------------------------------------------------------
Test Edit Project :: Create a Project and edit its name and descri... | PASS |
------------------------------------------------------------------------------
Test Copy Paste Dataset :: Create 2 Projects and a Dataset. Copy a... | PASS |
------------------------------------------------------------------------------
Test Copy Paste Image :: Copy Image from one dataset to another. C... | PASS |
------------------------------------------------------------------------------
Test Copy Paste Orphaned Image :: Test copy pasting an orphaned Im... | PASS |
------------------------------------------------------------------------------
Test Copy Paste Plate :: Test copy pasting a plate into another sc... | PASS |
------------------------------------------------------------------------------
Test Chgrp :: Tests chgrp of a Project to any other group             | PASS |
------------------------------------------------------------------------------
Test Drag And Drop Image                                              | PASS |
------------------------------------------------------------------------------
Test Drag and Drop Orphaned Image                                     | PASS |
------------------------------------------------------------------------------
Test Drag and Drop Orphaned Dataset                                   | PASS |
------------------------------------------------------------------------------
Test Drag and Drop Plate                                              | PASS |
------------------------------------------------------------------------------
Test Drag and Drop Dataset On Experimenter :: This test will curre... | PASS |
------------------------------------------------------------------------------
Test Drag and Drop Negative Tests                                     | PASS |
------------------------------------------------------------------------------
[Teardown]                                                            | PASS |
------------------------------------------------------------------------------
Post Test :: Tests ajax 'POST' updates.                               | PASS |
17 critical tests, 17 passed, 0 failed
17 tests total, 17 passed, 0 failed
==============================================================================
Output:  /Users/bramalingam/OME/openmicroscopy/components/tests/ui/target/reports/web/chrome/output.xml
Log:     /Users/bramalingam/OME/openmicroscopy/components/tests/ui/target/reports/web/chrome/log.html
Report:  /Users/bramalingam/OME/openmicroscopy/components/tests/ui/target/reports/web/chrome/report.html
```

**Comment by will-moore** [comment]
Several failures today https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-robotframework/159/robot/

**Comment by bramalingam** [comment]
current status : https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-robotframework/183/robot/report/log.html

**Comment by bramalingam** [comment]
@will-moore @aleksandra-tarkowska : All tests relating to this PR are passing now, can you please take a look and sign off on the PR.

File: components/tests/ui/testcases/web/post_test.txt [hunk 4]
```diff
@@ -114,3 +235,246 @@ Test Chgrp
...
+
+Test Drag and Drop Orphaned Dataset
+       
+    Wait Until Keyword Succeeds                 ${TIMEOUT}    ${INTERVAL}     Reload Page
+    Set Selenium Speed                          .3 
+    Select Experimenter                            
+    ${pid}=                                     Select First Project With Children
+    Select Experimenter                         
+    ${datasetId}                                Create Dataset
+    Wait Until Keyword Succeeds                 ${TIMEOUT}    ${INTERVAL}     Click Element                     refreshButton
```
Comment (will-moore) [inline components/tests/ui/testcases/web/post_test.txt]: What are you waiting for here? As soon as the button is clicked, the keyword has succeeded, right?
You have the same line in 8 places in total.

Comment (bramalingam) [inline components/tests/ui/testcases/web/post_test.txt]: Waiting for the refresh action to complete, somehow this allows the click element to complete its action as well..

**Comment by will-moore** [comment]
Seems that all these tests have been passing reliably over the last few days (3 of the last 4 days, ALL the  robot framework tests have passed https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-robotframework/188/).
So this is looking good to merge.

**Pull request merged**